addons/account/wizard/account_payment_register.py

```diff
@@ -266,13 +243,31 @@ def _compute_group_payment(self):
             else:
                 wizard.group_payment = False
 
-    @api.depends('company_id', 'source_currency_id')
+    @api.depends('company_id', 'source_currency_id', 'line_ids')
     def _compute_journal_id(self):
         for wizard in self:
             if wizard.journal_id:
                 wizard.journal_id = wizard.journal_id
             else:
-                wizard.journal_id = self._search_journal(wizard.company_id.id, wizard.source_currency_id.id)
+                partner_bank_id = wizard.line_ids.move_id.mapped('partner_bank_id')
+
+                company_domain = [('company_id', '=', wizard.company_id.id)]
+                bank_domain = [('bank_account_id', '=', partner_bank_id.id), ('type', '=', 'bank')] if len(partner_bank_id) == 1 else None
+                no_bank_domain = [('type', 'in', ('bank', 'cash'))]
+
+                journal = None
+                if wizard.source_currency_id:
+                    currency_domain = [('currency_id', '=', wizard.source_currency_id.id)]
+                    if bank_domain:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + bank_domain, limit=1)
+                    if not journal:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + no_bank_domain, limit=1)
+                if not journal and bank_domain:
+                    journal = self.env['account.journal'].search(company_domain + bank_domain, limit=1)
```

Comment:
I'm not sure which should have the highest priority; the currency or the bank.

Both make sense kind of sense so lets go with this.